engine/script/src/dmsdk/script/script.h

```diff
@@ -640,6 +640,15 @@ namespace dmScript
      */
     void UnrefInInstance(lua_State* L, int ref);
 
+    /*#
+     * Get the current game object URL
+     * @name CheckURL
+     * @param L [type:lua_State*] Lua state
+     * @param out_url [type:dmMessage::URL*] where to store the result
+     * @return result [type:bool] true if successful
+     */
+    bool CheckURL(lua_State* L, dmMessage::URL* out_url);
```

Comment:
Some api inconsistencies:
* The `CheckXxx` api should return the value type, which is (dmMessage::URL*) in this case.
* It should accept the `index` of the argument on the Lua stack
* Each type `Xxx`should have `IsXxx()`, `CheckXxx()`, `ToXxx()` and `PushXxx()`,

Note that we already have these functions, and they should only require to be moved, and update the documentation.

Reply:
Right, yeah, it is a good use case for getting the url of the current script.

> how does the user know when CheckURL failed?

The "CheckXxx" should always throw an error if they fail.
If we want to implement something more graceful, that's what we use the IsXxx/ToXxx for.